### ISSUE TITLE
Command line auto-completion for `daml`

### ISSUE THREAD

**Issue by circlespainter** [digital-asset/daml#3875]
It would be nice to have command line auto-completion for `daml` (even just commands, options as well would be even nicer).

**Comment by hurryabit** [comment]
@stefanobaghino-da The `optparse-applicative` library we're using the build the CLI can do this pretty much out of the box:
https://github.com/pcapriotti/optparse-applicative/wiki/Bash-Completion

**Comment by associahedron** [comment]
The PR above adds bash completions as part of `daml install`. Once the PR is merged and a release is made, you can get bash completions by running `daml install latest` twice (the first time installs the new assistant, the second time uses the new assistant to also install the bash completions).

This is only for bash, though. Windows and the latest version of Mac don't use bash. It would be a lot more work to add completions for other shells.
